core/trino-main/src/main/java/io/trino/execution/executor/PrioritizedSplitRunner.java

```diff
@@ -165,12 +164,10 @@ public ListenableFuture<Void> process()
             ListenableFuture<Void> blocked = split.processFor(SPLIT_RUN_QUANTA);
             CpuTimer.CpuDuration elapsed = timer.elapsedTime();
 
-            long endNanos = ticker.read();
-            long quantaScheduledNanos = endNanos - startNanos;
+            long quantaScheduledNanos = elapsed.getWall().roundTo(NANOSECONDS);
```

Comment:
This causes the creation of a duration object, and I'm not sure about the impact of that

Reply:
Actually, it doesn't. The objects are created by `CpuTimer.elapsedTime()` and passed into `CpuDuration`. The getter methods simply return the objects.